core/src/main/resources/error/error-classes.json

```diff
@@ -471,6 +471,11 @@
           "class <className> not found."
         ]
       },
+      "UNEXPECTED_INPUT_FOLDABLE_VALUE" : {
+        "message" : [
+          "Parameter <paramIndex> requires a foldable value of <requiredFoldable>, however <inputSql> has a foldable value of <inputFoldable>."
```

Comment:
From user's point of view, the message is not clear. What's the `foldable value of true`?

Can't you just re-use the existing error class: `NON_FOLDABLE_INPUT`?

Reply:
Hi @MaxGekk - I introduced the new error message as the text for `NON_FOLDABLE_INPUT` only makes sense when we expect the input to be foldable, but it is not. I tried to mimic the `ExpectsInputTypes.checkInputTypes` implementation in `ExpectsInputTypesAndFoldable`, which checks the foldable attribute of the expression against a user-provided expected value; this allows the trait to check if the foldable value is true OR false, and thus the error message needed `NON_FOLDABLE_INPUT` to be generalized. I wonder if we'll ever want to make sure that an input expression isn't foldable (IE for this function arg, you should only ever specify a column)? If not, then maybe I can update the `ExpectsInputTypesAndFoldable` to only check for cases where we expect an expression to be foldable, and then re-use the `NON_FOLDABLE_INPUT` error message.

Reply:
> when we expect the input to be foldable, but it is not

I wonder when we need to raise an error when the expected input non-foldable, but it is foldable actually. What's the case?

Reply:
I'm not sure if there's a clear use case where we'd want to check if the input is not foldable, but this implementation + the new error message does provide that level of flexibility.

Reply:
Yeah I think non-foldable might be enough, I can't think of a case where we would not want to allow a foldable input. This could also simplify the code a little bit in the checker since Seq[Option[Bool]] could become Seq[Bool]